diesel/src/pg/expression/functions.rs

```diff
@@ -748,3 +747,70 @@ define_sql_function! {
     /// ```
     fn array_append<Arr: ArrayOrNullableArray<Inner=T> + SingleValue, T: SingleValue>(a: Arr, e: T) -> Array<T>;
 }
+define_sql_function! {
+    /// Converts each array element to its text representation, and concatenates those elements
+    /// separated by the delimiter string. If `null_string` is given and is not `NULL`, then `NULL`
+    /// array entries are represented by that string; otherwise, they are omitted.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::array_to_string_with_null_string;
+    /// #     use diesel::sql_types::{Nullable, Text, Array};
+    /// #     let connection = &mut establish_connection();
+    ///
+    /// let result = diesel::select(array_to_string_with_null_string::<Array<_>, Nullable<Text>, _, _, _, _>(
+    ///     vec![Some("first"), None, Some("third")], ",", "NULL"))
+    ///     .get_result::<String>(connection)?;
+    ///
+    /// assert_eq!(result, "first,NULL,third");
+    /// #     Ok(())
+    /// # }
+    /// ```
+    fn array_to_string_with_null_string<Arr: ArrayOrNullableArray<Inner=Nullable<Text>> + SingleValue, T: SingleValue, N: SingleValue>(
+        a: Arr,
+        e: T,
+        n: N
+    ) -> Text;
+}
+
+#[cfg(feature = "postgres_backend")]
+define_sql_function! {
+    /// Converts each array element to its text representation, and concatenates those elements
+    /// separated by the delimiter string. This variant omits the `null_string` argument. See [`array_to_string_with_null_string`]
+    /// for a variant with that argument
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::array_to_string;
+    /// #     use diesel::sql_types::{Text, Array};
+    /// #     let connection = &mut establish_connection();
+    ///
+    /// let result = diesel::select(array_to_string::<Array<_>, Nullable<Text>, _, _>(
+    ///     vec![Some("first"), Some("second")], ","))
+    ///     .get_result::<String>(connection)?;
+    ///
+    /// assert_eq!(result, "first,second");
+    /// #     Ok(())
+    /// # }
+    /// ```
+    fn array_to_string<Arr: ArrayOrNullableArray<Inner=Nullable<Text>> + SingleValue, T: SingleValue>(
+        a: Arr,
+        e: T
+    ) -> Text;
```

Comment:
```suggestion
    fn array_to_string<Arr: ArrayOrNullableArray +SingleValue>(array: Arr, del: Text)) -> Text;
```

We do not need to make this function generic over the type of `del` here, as we only want to accept anything that is compatible with `Text` there. Therefore we can just write the sql side type (`Text`) there and do not need to care about generics. 
Also we want to accept any array type there, not only arrays of the type `Text`, which means we do not care about the `Inner` associated type here.